scipy/spatial/transform/_rotation.pyx

```diff
@@ -1835,6 +2101,150 @@ cdef class Rotation:
+            The returned angles are in the range:
+
+            - First angle belongs to [-180, 180] degrees (both inclusive)
+            - Third angle belongs to [-180, 180] degrees (both inclusive)
+            - Second angle belongs to a set of size 180 degrees, 
+              given by: ``[-abs(lambda), 180 - abs(lambda)]``, where ``lambda``
+              is the angle between the first and third axes.
+
+        References
+        ----------
+        .. [1] Shuster, Malcolm & Markley, Landis. (2003). Generalization of
+               the Euler Angles. Journal of the Astronautical Sciences. 51. 123-132. 10.1007/BF03546304.
+        .. [2] Bernardes E, Viollet S (2022) Quaternion to Euler angles
+               conversion: A direct, general and computationally efficient method.
+               PLoS ONE 17(11): e0276302. 10.1371/journal.pone.0276302
+        .. [3] https://en.wikipedia.org/wiki/Gimbal_lock#In_applied_mathematics
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+
+        Davenport angles are a generalization of Euler angles, when we use the
+        canonical basis axes:
+
+        >>> ex = [1, 0, 0]
+        >>> ey = [0, 1, 0]
+        >>> ez = [0, 0, 1]
+
+        Represent a single rotation:
+
+        >>> r = R.from_rotvec([0, 0, np.pi/2])
+        >>> r.as_davenport([ez, ex, ey], 'extrinsic', degrees=True)
+        array([90.,  0.,  0.])
+        >>> r.as_euler('zxy', degrees=True)
+        array([90.,  0.,  0.])
+        >>> r.as_davenport([ez, ex, ey], 'extrinsic', degrees=True).shape
+        (3,)
+
+        Represent a stack of single rotation:
+
+        >>> r = R.from_rotvec([[0, 0, np.pi/2]])
+        >>> r.as_davenport([ez, ex, ey], 'extrinsic', degrees=True)
+        array([[90.,  0.,  0.]])
+        >>> r.as_davenport([ez, ex, ey], 'extrinsic', degrees=True).shape
+        (1, 3)
+
+        Represent multiple rotations in a single object:
+
+        >>> r = R.from_rotvec([
+        ... [0, 0, 90],
+        ... [45, 0, 0]], degrees=True)
+        >>> r.as_davenport([ez, ex, ey], 'extrinsic', degrees=True)
+        array([[90.,  0.,  0.],
+               [ 0., 45.,  0.]])
+        >>> r.as_davenport([ez, ex, ey], 'extrinsic', degrees=True).shape
+        (2, 3)
+        """
+        if order in ['e', 'extrinsic']:
+            extrinsic = True
+        elif order in ['i', 'intrinsic']:
+            extrinsic = False
+        else:
+            raise ValueError("order should be 'e'/'extrinsic' for extrinsic "
+                             "sequences or 'i'/'intrinsic' for intrinsic "
+                             "sequences, got {}".format(order))
+
+        if len(axes) != 3:
+            raise ValueError("Expected 3 axes, got {}.".format(len(axes)))
+
+        for axis in axes:
+            if len(axes) != 3:
+                raise ValueError("Axes must be vectors of length 3.")
```

Comment:
Should it be changed into shape check of the `axes` array (after `asarray` is applied)?

Reply:
Changed, what do you think now?